### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by dovanduy109** [koalaman/shellcheck#1798]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2039, SC2039, SC2039, SC2039, SC2007, SC2039, SC2154, SC2039
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
## Example: The shebang says 'sh' so shellcheck warns about portability
##          Change it to '#!/bin/bash' to allow bashisms
for n in {1..$RANDOM}
do
  str=""
  if (( n % 3 == 0 ))
  then
    str="fizz"
  fi
  if [ $[n%5] == 0 ]
  then
    str="$strbuzz"
  fi
  if [[ ! $str ]]
  then
    str="$n"
  fi
  echo "$str"
done

```

#### Here's what shellcheck currently says:
```
Line 4	SC2039: In POSIX sh, brace expansion is undefined.
Line 4	SC2039: In POSIX sh, RANDOM is undefined.
Line 7	SC2039: In POSIX sh, standalone ((..)) is undefined.
Line 11	SC2039: In POSIX sh, $[..] in place of $((..)) is undefined.
Line 11	SC2007: Use $((..)) instead of deprecated $[..]
Line 11	SC2039: In POSIX sh, == in place of = is undefined.
Line 13	SC2154: strbuzz is referenced but not assigned.
Line 15	SC2039: In POSIX sh, [[ ]] is undefined.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by Nightfirecat** [comment]
What is the bug in what you've posted? These seem like valid warnings to me.

**Comment by stdedos** [comment]
Duplicate of #1797
